### PR TITLE
feat(UI): Add "Getting Started" Modal on fresh deployment

### PR DESCRIPTION
**Pull request by jjoyce0510**
This introduces a "Getting Started" modal that shows when no metadata has been ingested into your DataHub instance. Additionally, we inform the user on the home page when the instance does not have metadata.  See screenshots below! 

Screen Shot 2021-12-19 at 6 22 23 PM
Screen Shot 2021-12-19 at 6 22 30 PM
Screen Shot 2021-12-19 at 6 20 26 PM

Also renamed the links in the top-right drop down to remove unnecessary DataHub prefix. And added "releases" to the mix.
Screen Shot 2021-12-19 at 6 37 25 PM

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-web-react/src/app/home/HomePageRecommendations.tsx [hunk 1]
```diff
@@ -75,25 +85,58 @@ export const HomePageRecommendations = ({ userUrn }: Props) => {
...
+                                    <div
+                                        style={{
+                                            display: 'flex',
+                                            flexDirection: 'column',
+                                            justifyContent: 'center',
+                                            alignItems: 'center',
+                                        }}
+                                    >
+                                        <NoMetadataEmpty description="No Metadata Found 😢" />
+                                        <Button style={{ margin: 16 }} onClick={() => setShowGettingStartedModal(true)}>
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/home/HomePageRecommendations.tsx:132]: 😏

File: datahub-web-react/src/app/home/HomePageRecommendations.tsx [hunk 2]
```diff
@@ -75,25 +85,58 @@ export const HomePageRecommendations = ({ userUrn }: Props) => {
...
+                                            key={entityCount.entityType}
+                                            entityType={entityCount.entityType}
+                                            count={entityCount.count}
+                                        />
+                                    ),
+                            )}
+                        </BrowseCardContainer>
+                    ) : (
+                        <List
+                            locale={{
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/home/HomePageRecommendations.tsx:121]: why is this a list?

File: datahub-web-react/src/app/home/HomePageRecommendations.tsx [hunk 3]
```diff
@@ -75,25 +85,58 @@ export const HomePageRecommendations = ({ userUrn }: Props) => {
     });
     const recommendationModules = data?.listRecommendations?.modules;
 
+    // Determine whether metadata has been ingested yet.
+    const hasLoadedEntityCounts = orderedEntityCounts && orderedEntityCounts.length > 0;
+    const hasIngestedMetadata =
+        orderedEntityCounts && orderedEntityCounts.filter((entityCount) => entityCount.count > 0).length > 0;
+
+    useEffect(() => {
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/home/HomePageRecommendations.tsx:93]: nice useEffect!!!

**Review by gabe-lyons — APPROVED** [review]

File: datahub-web-react/src/app/home/HomePageRecommendations.tsx (hunk 1 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/home/HomePageRecommendations.tsx:132]: :(

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**